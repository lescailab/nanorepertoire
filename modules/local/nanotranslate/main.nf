process NANOTRANSLATE {
    tag "$meta.id"
    label 'process_high'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/biopython:1.78' :
        'ghcr.io/nibscbioinformatics/biopython:v1.78' }"

    input:
    tuple val(meta), path(mergedfq)

    output:
    tuple val(meta), path('*.fasta'), emit: fasta
    tuple val(meta), path('*.log')  , emit: log
    path 'versions.yml'             , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    gzip -cd ${mergedfq} >sequences.fastq
    translate.py \
    sequences.fastq \
    ${meta.id}_AA_translated.fasta \
    >${meta.id}_AA_translated.log

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        biopython: \$(python -c "import Bio; print(Bio.__version__)")
    END_VERSIONS
    """
}

