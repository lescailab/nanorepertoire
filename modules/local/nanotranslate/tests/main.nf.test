nextflow_process {
    name "Test Process NANOTRANSLATE"
    script "../main.nf"
    process "NANOTRANSLATE"
    config "./nextflow.config"
    tag "modules"
    tag "modules_nfcore"
    tag "flash"
    tag "nanotranslate"

    test("nanotranslate-test") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test1', single_end:false ], // meta map
                    [file("https://github.com/lescailab/testdata/raw/master/nanobodies/test01_flash_merged.fastq.gz", checkIfExists: true)
                    ]
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
