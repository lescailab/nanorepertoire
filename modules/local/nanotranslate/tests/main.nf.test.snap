{
    "nanotranslate-test": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test1",
                            "single_end": false
                        },
                        "test1_AA_translated.fasta:md5,c7d4ca2158a40a3543b54970d30a861d"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test1",
                            "single_end": false
                        },
                        [
                            "test1_AA_translated.fasta.log:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test1_AA_translated.log:md5,984b04b16616f50e9d43d49afd604138"
                        ]
                    ]
                ],
                "2": [
                    "versions.yml:md5,8b1c036b46795ec5f9f5986280401944"
                ],
                "fasta": [
                    [
                        {
                            "id": "test1",
                            "single_end": false
                        },
                        "test1_AA_translated.fasta:md5,c7d4ca2158a40a3543b54970d30a861d"
                    ]
                ],
                "log": [
                    [
                        {
                            "id": "test1",
                            "single_end": false
                        },
                        [
                            "test1_AA_translated.fasta.log:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test1_AA_translated.log:md5,984b04b16616f50e9d43d49afd604138"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,8b1c036b46795ec5f9f5986280401944"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-08T10:31:53.063863"
    }
}